docs/DataFetchingGuide.md

```diff
@@ -0,0 +1,415 @@
+For instance, to query the API for a single record, react-admin calls `dataProvider.getOne()`:
+
+```tsx
+const response = await dataProvider.getOne('posts', { id: 123 });
+console.log(response.data); // { id: 123, title: "hello, world" }
+```
+
+The Data Provider is responsible for transforming these method calls into HTTP requests and converting the responses into the format expected by react-admin. In technical terms, a Data Provider is an *adapter* for an API.
+
+A Data Provider must implement the following methods:
+
+```jsx
+const dataProvider = {
+    async getList(resource, { sort, filter, pagination }) => ({ data: Record[], total: number }), 
+    async getOne(resource, { id }) => ({ data: Record }),
+    async getMany(resource, { ids }) => ({ data: Record[] }),
+    async getManyReference(resource, { target, id, sort, filter, pagination }) => ({ data: Record[], total: number }), 
+    async create(resource, { data }) => ({ data: Record }),
+    async update(resource, { id, data }) => ({ data: Record }),
+    async updateMany(resource, { ids, data }) => ({ data: Identifier[] }),
+    async delete(resource, { id } ) => ({ data: Record }),
+    async deleteMany(resource, { ids }) => ({ data: Identifier[] }),
+}
+```
+
+**Tip**: A Data Provider can have [additional methods](#adding-custom-methods) beyond these 9. For example, you can add custom methods for non-REST API endpoints, tree structure manipulations, or realtime updates.
+
+The Data Provider is a key part of react-admin's architecture. By standardizing the Data Provider interface, react-admin can offer powerful features, like reference handling, optimistic updates, and autogenerated CRUD components.
+
+## Backend Agnostic
+
+Thanks to this adapter system, react-admin can communicate with any API. It doesn't care if your API is a REST API, a GraphQL API, a SOAP API, a JSON-RPC API, or even a local API. It doesn't care if your API is written in PHP, Python, Ruby, Java, or JavaScript. It doesn't care if your API is a third-party API or a homegrown API.
+
+React-admin ships with [more than 50 data providers](./DataProviderList.md) for popular API flavors.
+
+You can also [write your own Data Provider](./DataProviderWriting.md) to fit your backend's particularities. Data Providers can use `fetch`, `axios`, `apollo-client`, or any other library to communicate with APIs. The Data Provider is also the ideal place to add custom HTTP headers, authentication, etc.
+
+Check out the [Data Provider Setup](./DataProviders.md) documentation for more details on how to set up a Data Provider in your app.
+
+## Calling The Data Provider
+
+Many react-admin components use the Data Provider: page components like `<List>` and `<Edit>`, reference components like `<ReferenceField>` and `<ReferenceInput>`, action Buttons like `<DeleteButton>` and `<SaveButton>`, and many more.
+
+If you need to call the Data Provider directly from your components, you can use the specialized hooks provided by react-admin:
```

Comment:
I would mention that most of the time there are components that are better suited (base components, etc.)

Reply:
This would add too much information at taht stage.